### ISSUE TITLE
Dynamic TEXT Section Image Verification

### ISSUE THREAD

**Issue by friuns2** [LordNoteworthy/al-khaser#139]
Checking code for modifications in memory but how reliable it is?

Here is article i tried compile code it working for some extent 
https://www.codeproject.com/Articles/17636/Dynamic-TEXT-Section-Image-Verification

**Comment by gsuberland** [comment]
Yes, this is something we'll add at some point. I'll be adding PE parsing functionality in soon as part of another feature anyway, so we might as well take advantage of it. We'll probably only target the main module rather than all DLLs in the process as it's less likely to flag up false positives that way.

We should avoid using hashing as it needlessly overcomplicates the check; we can just check byte-for-byte. This also allows us to more easily apply relocation fixups and avoid the ASLR issues that the author ran into.

**Comment by friuns2** [comment]
found another implementation
https://github.com/rootm0s/Protectors/blob/master/AntiDebugging/Sources/TextSectionHasher.cpp
